presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -342,21 +345,31 @@ public PlanWithProperties visitSemiJoin(SemiJoinNode node, Void context)
 
             // make filtering source match requirements of source
             PlanNode filteringSourceNode;
-            if (source.getProperties().isPartitioned()) {
-                filteringSourceNode = new ExchangeNode(
-                        idAllocator.getNextId(),
-                        ExchangeNode.Type.REPLICATE,
-                        ImmutableList.of(),
-                        Optional.<Symbol>empty(),
-                        ImmutableList.of(filteringSource.getNode()),
-                        filteringSource.getNode().getOutputSymbols(),
-                        ImmutableList.of(filteringSource.getNode().getOutputSymbols()));
+            if (distributedJoins) {
+                List<Symbol> sourceSymbols = Collections.singletonList(node.getSourceJoinSymbol());
```

Comment:
Usually we use `ImmutableLists.of()` rather than `Collections.singletonList()`. Also, it looks like the Lists are only used once, so I would just inline the construction of them